semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -3056,6 +3057,85 @@ and m_directive a b =
   | G.OtherDirective _ ->
       fail ()
 
+and m_directive_vs_def a b =
+  let f (filea, importsa) = function
+    (* const x = require('y');
+     *
+     * This is for JS but there is no need to gate on language, since the
+     * `Require` node is unique to JS. *)
+    | ( { B.name = B.EN (B.Id (_id, _id_info)); _ },
+        B.VarDef
+          {
+            vinit =
+              Some
+                {
+                  e =
+                    B.Call
+                      ( { e = B.IdSpecial (B.Require, _); _ },
+                        (_, [ B.Arg { e = B.L (B.String fileb); _ } ], _) );
+                  _;
+                };
+            vtype = _;
+          } )
+      when importsa = [] ->
+        (* Match the pattern `import "foo"` against `const x = require("foo")` *)
+        m_wrap m_string filea fileb
+    (* const {x, y} = require('z');
+     *
+     * This is for JS but there is no need to gate on language, since the
+     * `Require` node is unique to JS. *)
+    | ( { B.name = B.EN (B.Id ((id_str, _), _)); _ },
+        B.VarDef
+          {
+            vinit =
+              Some
+                {
+                  e =
+                    B.Assign
+                      ( { e = B.Record (_, fields, _); _ },
+                        _,
+                        {
+                          e =
+                            B.Call
+                              ( { e = B.IdSpecial (Require, _); _ },
+                                (_, [ B.Arg { e = B.L (B.String fileb); _ } ], _)
+                              );
+                          _;
+                        } );
+                  _;
+                };
+            vtype = _;
+          } )
+      when id_str = B.special_multivardef_pattern ->
+        let* () = m_wrap m_string filea fileb in
+        m_list_in_any_order ~less_is_ok:true m_import_vs_field importsa fields
+    | _ -> fail ()
+  in
+  match a.d with
+  (* JS: `import {x, y as z} from 'a';` *)
+  | G.ImportFrom (_, G.FileName filea, importsa) -> f (filea, importsa) b
+  (* JS: `import 'a';` *)
+  | G.ImportAs (_, G.FileName filea, None) -> f (filea, []) b
+  | _ -> fail ()
+
+and m_import_vs_field a b =
```

Comment:
Maybe gate this under javascript or add a comment specifying that? It's pretty close to the use, so not a big deal, just thinking that if we ever find another language with similar semantics it might not do it the same way